CHANGELOG.md

```diff
@@ -6,6 +6,73 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ## [Unreleased]
 
+## [0.16.0] - 2022-02-01
+
```

Comment:
I wonder whether it's worth a brief message at the top like the following?

```
*Note*: This is a significant release which introduces support for V14 metadata and macro based codegen, as well as making many breaking changes to the API.
```

Reply:
Good pt, adding.